config/index.js

```diff
@@ -0,0 +1,116 @@
+let _        = require("lodash"),
+    contract = require("truffle-contract"),
+    nconf    = require("nconf"),
+    path     = require("path"),
+    Web3     = require("web3"),
+    defaults = require("./defaults.json");
+
+let configPath = path.join(__dirname, '../config');
+nconf.env().argv();
+nconf.file({file: path.join(configPath, (nconf.get('NODE_ENV') || "dev") + ".json")});
+nconf.defaults(defaults);
+
+let provider = new Web3.providers.HttpProvider(nconf.get("ethereum:provider"));
+let web3     = new Web3(provider);
+
+let properties       = ["contract_name", "abi", "unlinked_binary"];
+let contractDefaults = {
+    from: web3.eth.accounts[0],
+    gas: 4712388,
+    gasPrice: 100000000000
+};
+
+let MathLib = contract(_.pick(require("../build/contracts/MathLib.json"), properties));
+MathLib.setProvider(provider);
+MathLib.defaults(contractDefaults);
+
+let GroveLib = contract(_.pick(require("../build/contracts/GroveLib.json"), properties));
+GroveLib.setProvider(provider);
+GroveLib.defaults(contractDefaults);
+
+let IterTools = contract(_.pick(require("../build/contracts/IterTools.json"), properties));
+IterTools.setProvider(provider);
+IterTools.defaults(contractDefaults);
+
+let ExecutionLib = contract(_.pick(require("../build/contracts/ExecutionLib.json"), properties));
+ExecutionLib.setProvider(provider);
+ExecutionLib.defaults(contractDefaults);
+
+let Digger = contract(_.pick(require("../build/contracts/Digger.json"), properties));
```

Comment:
FYI: This contract should not longer be necessary since stack depth checking has been made obsolete.